java-manta-it/pom.xml

```diff
@@ -112,8 +112,181 @@
+                    <execution>
+                        <id>add-shade-workaround-source</id>
+                        <phase>generate-sources</phase>
+                        <goals>
+                            <goal>add-source</goal>
+                        </goals>
+                        <configuration>
+                            <sources>
+                                <source>src/shade-workaround</source>
+                            </sources>
+                        </configuration>
+                    </execution>
+                </executions>
+            </plugin>
+            <plugin>
+                <artifactId>maven-resources-plugin</artifactId>
+                <version>${maven-maven-resources-plugin.version}</version>
+                <executions>
+                    <execution>
+                        <id>copy-shade-workaround-src</id>
+                        <phase>pre-integration-test</phase>
+                        <goals>
+                            <goal>copy-resources</goal>
+                        </goals>
+                        <configuration>
+                            <outputDirectory>src/shade-workaround</outputDirectory>
+                            <resources>
+                                <resource>
+                                    <directory>${multi.module.root}/java-manta-client-unshaded/src/main/java</directory>
+                                </resource>
+                            </resources>
+                        </configuration>
+                    </execution>
+                </executions>
+            </plugin>
+            <plugin>
+                <!-- unpack the modified-by-relocation classes from
+                     the client jar -->
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-dependency-plugin</artifactId>
+                <version>${maven-build-helper-maven-plugin.version}</version>
+                <executions>
+                    <execution>
+                        <id>unpack-relocated-classes</id>
+                        <phase>pre-integration-test</phase>
+                        <goals>
+                            <goal>unpack</goal>
+                        </goals>
+                        <configuration>
+                            <artifactItems>
+                                <artifactItem>
+                                    <groupId>com.joyent.manta</groupId>
+                                    <artifactId>java-manta-client</artifactId>
+                                    <type>jar</type>
+                                    <overWrite>true</overWrite>
+                                    <outputDirectory>${project.build.directory}/classes</outputDirectory>
+                                    <includes>**/*.class</includes>
+                                    <excludes>com/joyent/manta/com/**,com/joyent/manta/org/**,com/joyent/manta/io/**,com/joyent/manta/http/signature/**</excludes>
```

Comment:
Just out of curiosity, under what circumstances would this `<excludes>` list change? They seem to be the relocation roots but I just want to be sure I know how to maintain this. I'm assuming coverage for library classes shows up if this isn't set?

Reply:
Yep, this is just working backwards from the relocation roots in `java-manta-client/pom.xml`.  Additions there may also require changes here.  At some point (4.0?) I think it would be worthwhile to have only a single package path for relocated (ex `com.joyent.manta.shade`) just to keep things a bit tidier.

Reply:
Agreed about the single package relocation. The question of "why isn't everything shaded to the same place?" has crossed my mind once before. I'll file an issue for this.